controllers/openstackcluster_controller.go

```diff
@@ -236,6 +209,36 @@ func contains(arr []string, target string) bool {
 	return false
 }
 
+func deleteBastion(log logr.Logger, osProviderClient *gophercloud.ProviderClient, clientOpts *clientconfig.ClientOpts, openStackCluster *infrav1.OpenStackCluster) error {
+	computeService, err := compute.NewService(osProviderClient, clientOpts, log)
+	if err != nil {
+		return err
+	}
+	networkingService, err := networking.NewService(osProviderClient, clientOpts, log)
+	if err != nil {
+		return err
+	}
+
+	if bastion := openStackCluster.Status.Bastion; bastion != nil {
```

Comment:
I wouldn't write this in an additional var. That's usually only done for errors or stuff like map lookups in Golang. So please just use `openStackCluster.Status.Bastion`